### ISSUE TITLE
[a11y]: HeaderSideNavItems should use a ul rather than div

### ISSUE THREAD

**Issue by samrobbins85** [carbon-design-system/carbon#10515]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

7.50.0

### React version

16.14.0

### Automated testing tool and ruleset

axe devtools

### Description

Using the method described in storybook of `<HeaderSideNavItems>` with `<HeaderMenuItem>`s nested inside you get a div (HeaderSideNavItems) containing `li`s (HeaderMenuItem). Instead HeaderSideNavItems should be a list 

### WCAG 2.1 Violation

Success Criterion 1.3.1 Info and Relationships

### CodeSandbox example

https://codesandbox.io/s/ibm-sidenav-a11y-2-yrwlo?file=/package.json

### Steps to reproduce

Narrow the window such that the mobile menu comes up, open this menu and inspect the top section, it contains a `div` containing `li`s

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
